server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -2605,7 +2605,7 @@ public void updatePoolsFromStackWithoutDeletingStack(Consumer consumer, List<Poo
         List<Pool> filteredPools = new LinkedList<>();
         for (Pool p : pools) {
             ValidationResult result = activationKeyRules.runPoolValidationForActivationKey(key, p, null);
-            if (result.isSuccessful() && (!result.hasWarnings() || includeWarnings)) {
+            if (!result.hasErrorKeys() && (!result.hasWarnings() || includeWarnings)) {
```

Comment:
The issue here was that a new piece of data (`errorKeys`) was added to ValidationResult in addition to `errors`, which is not taken into account by the `isSuccessful` method. This will work, but I think to future-proof the use of the `isSuccessful` method (which is used elsewhere, and in the future when we move more code from rules.js to Java, the same bug will appear there too), we should leave this check as it was and simply alter ValidationResult.isSuccessful to more correctly check both errors and errorKeys as such: `return !hasErrors() && !hasErrorKeys();`

Reply:
Done.

Reply:
Is there a reason we don't just fix ```isSuccessful``` and any places that break as a consequence of that fix? Seems like a trap to leave a method like that around that doesn't check what its name implies.

Reply:
The method `isSuccessful()` is being updated with `return !hasErrors() && !hasErrorKeys();`. Also the above change in `CandlepinPoolManager.java` is reverted. Is there anything else that needs to be done?